### ISSUE TITLE
AbortAddon: `setTimeout` doesn't work in combination with own `AbortController` instance

### ISSUE THREAD

**Issue by adriaanmeuris** [elbywan/wretch#256]
Timeouts and aborting requests work fine when using `.controller`:
```js
// Aborts after 1 second timeout
const [controller, w] = wretch(url).addon(AbortAddon()).get().setTimeout(1000).controller();

// Aborts after 0.5 second using `.abort`
const [controller, w] = wretch(url).addon(AbortAddon()).get().setTimeout(1000).controller();  
setTimeout(() => { controller.abort(); }, 500);
```

However when you want to provide a controller yourself, the implementation is broken:
```js
// Aborts after 1 second timeout, but can't be aborted before because we don't provide a controller
await wretch(url).addon(AbortAddon()).get().setTimeout(1000);

// Doesn't abort after 0.5 second timeout (even though configured so), but can be aborted using `.abort`
const controller = new AbortController();
await wretch(url).addon(AbortAddon()).signal(controller).get().setTimeout(500);
setTimeout(() => { controller.abort(); }, 1000);
```

Additionally, I don't this it's currently possible to differentiate between a configured timeout error, and manually aborting it (they both trigger the `onAborted` catcher)

**Comment by elbywan** [comment]
Hey @adriaanmeuris,

```ts
// Doesn't abort after 0.5 second timeout (even though configured so)
await wretch(url).addon(AbortAddon()).signal(controller).get().setTimeout(500);
```

I believe you have to explicitly pass your custom controller as an argument: `.setTimeout(500, controller)`

> If you use a custom AbortController associated with the request, pass it as the second argument.

https://elbywan.github.io/wretch/api/interfaces/addons_abort.AbortResolver.html#setTimeout

> I don't this it's currently possible to differentiate between a configured timeout error, and manually aborting it

They are both equivalent indeed, since `setTimeout` calls `controller.abort()` under the hood.

**Comment by adriaanmeuris** [comment]
Thanks for clarifying. Is `.signal(controller)` still required if the controller is passed as a second argument to `setTimeout`?

I need to distinguish manual aborts (e.g. aborting irrelevant requests before they resolve) from actual timeouts, especially for error reporting. Any advice on this, or would you consider a PR that allow to explicitly separate the two cases?

**Comment by elbywan** [comment]
> Is .signal(controller) still required if the controller is passed as a second argument to setTimeout?

Yes, otherwise the controller will not be associated with the request.

> I need to distinguish manual aborts (e.g. aborting irrelevant requests before they resolve) from actual timeouts, especially for error reporting. Any advice on this (…)

There is a way to specify a reason when aborting, maybe you could leverage that?

Something like (reusing your example): 

```ts
const controller = new AbortController();
setTimeout(() => { controller.abort("custom reason"); }, 250);
try {
  await wretch(url).addon(AbortAddon()).signal(controller).get().setTimeout(500, controller);
} catch (error) {
  if (controller.signal.reason === "custom reason") {
    // aborted manually
  } else if (controller.signal.aborted) {
    // aborted due to the timeout
  } else {
    // … //
  }
}
```
